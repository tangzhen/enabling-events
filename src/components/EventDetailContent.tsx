import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import React from "react";
import clsx from "clsx";
import moment from "moment";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import FacebookIcon from "@material-ui/icons/Facebook";
import TwitterIcon from "@material-ui/icons/Twitter";
import EmailIcon from "@material-ui/icons/Email";
import EnablingEvent from "../model/EnablingEvent";

interface EventDetailContentProps {
  event: EnablingEvent;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    description: {
      padding: theme.spacing(4),
    },
    about: {
      marginBottom: theme.spacing(4),
    },
    extraInfo: {
      paddingTop: theme.spacing(4),
    },
    bold: {
      fontWeight: "bold",
    },
    sectionPadding: {
      paddingTop: theme.spacing(10),
    },
  })
);

export default function EventDetailContent({ event }: EventDetailContentProps) {
  const classes = useStyles();

  return (
    <Grid container>
      <Grid item xs={8} className={classes.description}>
        <Typography variant="h6" className={classes.about}>
          About this Event
        </Typography>
        <Typography>{event.summary}</Typography>

        <Typography className={clsx(classes.bold, classes.sectionPadding)}>
          Share with Friends
        </Typography>
        <FacebookIcon color="primary" style={{ fontSize: 30 }} />
        <TwitterIcon color="primary" style={{ fontSize: 30, paddingLeft: 5 }} />
        <EmailIcon color="primary" style={{ fontSize: 30, paddingLeft: 5 }} />
      </Grid>
      <Grid item xs={4} className={classes.extraInfo}>
        <Typography className={classes.bold}>Date and Time</Typography>
        <Typography>
          {moment(event.startDate).format("ddd, MMMM D, YYYY")}
        </Typography>
        <Typography>3.00 PM - 6.00 PM</Typography>
        <Button color="primary">Add to Calendar</Button>

        <Typography className={clsx(classes.bold, classes.sectionPadding)}>
          Location
        </Typography>
        <Typography>{event.location}</Typography>

        <Typography className={clsx(classes.bold, classes.sectionPadding)}>
          Refund Policy
        </Typography>
        <Typography>Contact the organizer to request a refund</Typography>
      </Grid>
    </Grid>
  );
}
