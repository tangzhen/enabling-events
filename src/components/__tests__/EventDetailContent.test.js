import React from "react";
import { render } from "../../../test/test-utils";
import EventDetailContent from "../EventDetailContent";

describe("EventDetailContent", () => {
  it("should render component correct", () => {
    const event = {
      id: 1,
      title: "Pancake With Sliced Strawberry",
      bg: "/files/1.jpg",
      startDate: "2020-10-24T08:52:10+08:00",
      endDate: "2020-10-24T08:52:10+08:00",
      organizer: "Standard Chartered",
      location: "Entrance of National Gallery",
      summary:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis, lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo, vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis enim lobortis scelerisque fermentum dui faucibus in ornare quam viverra orci sagittis eu volutpat odio facilisis mauris sit amet massa vitae tortor condimentum.",
    };
    const { container } = render(<EventDetailContent event={event} />);
    expect(container).toMatchSnapshot();
  });
});
