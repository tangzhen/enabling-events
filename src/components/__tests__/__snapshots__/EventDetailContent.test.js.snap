// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetailContent should render component correct 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-description-1 MuiGrid-item MuiGrid-grid-xs-8"
    >
      <h6
        class="MuiTypography-root makeStyles-about-2 MuiTypography-h6"
      >
        About this Event
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis, lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo, vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis enim lobortis scelerisque fermentum dui faucibus in ornare quam viverra orci sagittis eu volutpat odio facilisis mauris sit amet massa vitae tortor condimentum.
      </p>
      <p
        class="MuiTypography-root makeStyles-bold-4 makeStyles-sectionPadding-5 MuiTypography-body1"
      >
        Share with Friends
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
        focusable="false"
        style="font-size: 30px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
        focusable="false"
        style="font-size: 30px; padding-left: 5px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
        focusable="false"
        style="font-size: 30px; padding-left: 5px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
        />
      </svg>
    </div>
    <div
      class="MuiGrid-root makeStyles-extraInfo-3 MuiGrid-item MuiGrid-grid-xs-4"
    >
      <p
        class="MuiTypography-root makeStyles-bold-4 MuiTypography-body1"
      >
        Date and Time
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Sat, October 24, 2020, 8.52 AM
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        to
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Sat, October 24, 2020, 11.12 AM
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add to Calendar
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <p
        class="MuiTypography-root makeStyles-bold-4 makeStyles-sectionPadding-5 MuiTypography-body1"
      >
        Location
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Entrance of National Gallery
      </p>
      <p
        class="MuiTypography-root makeStyles-bold-4 makeStyles-sectionPadding-5 MuiTypography-body1"
      >
        Refund Policy
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Contact the organizer to request a refund
      </p>
    </div>
  </div>
</div>
`;
